tripy/tripy/frontend/utils.py

```diff
@@ -19,43 +19,40 @@
 import inspect
 import numbers
 from collections import deque
-from typing import List, Sequence, Set, Union
+from typing import List, Set, Union
 
 from tripy import utils
 from tripy.common.exception import raise_error
 from tripy.flat_ir.function import FlatIRFunction
 from tripy.flat_ir.ops import BaseFlatIROp
+from tripy.types import ShapeLike, TensorLike
 
 
-# Try to include correct column offsets for non-tensor arguments.
-def _add_column_info_for_non_tensor(
-    arg,
-    arg_index,
-    is_kwarg,
-    dtype,
-    num_args,
-    func_name,
-    skip_num_stack_entries,
-    list_index=None,
-    TensorType=None,
-):
-    from tripy.common.datatype import bool as tp_bool
-    from tripy.common.datatype import floating, integer
-    from tripy.frontend.shape import Shape
+def tensor_from_shape_like(arg: ShapeLike) -> "tripy.Tensor":
+    from tripy.common.datatype import int32
+    from tripy.frontend.dimension_size import DimensionSize
     from tripy.frontend.tensor import Tensor
-    from tripy.frontend.trace.ops.cast import cast
+    from tripy.frontend.trace.ops.concatenate import concatenate
+    from tripy.frontend.trace.ops.reshape import Reshape
+
+    if not arg:
+        return Tensor([], dtype=int32)
+
```

Comment:
Might worth adding a simple optimization:
```py
if not arg or all(isinstance(e, int) for e in arg):
    return Tensor(arg, dtype=int32)
```

Reply:
@yizhuoz004 Thanks, I implemented a generalized version of this approach which will accumulate contiguous sequences of integers instead of creating individual tensors for each.